src/config.py

```diff
@@ -121,10 +160,20 @@ class OrderbookConfig:
     barn_db_url: str
 
     @staticmethod
-    def from_env() -> OrderbookConfig:
+    def from_network(network: Network) -> OrderbookConfig:
         """Initialize orderbook config from environment variables."""
-        prod_db_url = os.environ.get("PROD_DB_URL", "")
-        barn_db_url = os.environ.get("BARN_DB_URL", "")
+        match network:
+            case Network.MAINNET:
+                prod_db_url = os.environ.get("PROD_DB_URL_MAINNET", "")
+                barn_db_url = os.environ.get("BARN_DB_URL_MAINNET", "")
+            case Network.GNOSIS:
+                prod_db_url = os.environ.get("PROD_DB_URL_GNOSIS", "")
+                barn_db_url = os.environ.get("BARN_DB_URL_GNOSIS", "")
+            case Network.ARBITRUM_ONE:
+                prod_db_url = os.environ.get("PROD_DB_URL_ARBITRUM", "")
+                barn_db_url = os.environ.get("BARN_DB_URL_ARBITRUM", "")
```

Comment:
Came here from https://github.com/cowprotocol/infrastructure/pull/2349.
This seems like it complicates the configuration. Why not continue to always read the URLs from `PROD_DB_URL` and `BARN_DB_URL` and expect the job to be run with the correctly configured env variables?

Same comment applies to `NODE_URL`.

Reply:
In principle it is possible to have one config file with all environment variables per chain, and remove all post-fixes. With our infrastructure it would be one config file and one secret file per chain then, right?

At some point we had planned to compute all payments in one run and then all secrets would need to be available from the same environment. But this is not planned anymore.

Reply:
> With our infrastructure it would be one config file and one secret file per chain then, right?

Yeah, I think so. Both approaches can definitely work. Just seemed surprising to me that you'd basically have 2 places where you branch based on the network. First when you decide which ENV variable to write your values into and later when you decide which ENV variable to read from. 🤷‍♂️